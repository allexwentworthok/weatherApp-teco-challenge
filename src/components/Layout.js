import React from 'react'
import { StyleSheet, KeyboardAvoidingView, View, useWindowDimensions } from 'react-native'


export default function Layout({ children }) {
const { width, height } = useWindowDimensions();
    const styles = StyleSheet.create({
        container: {
            paddingHorizontal: width * 0.05,
            paddingTop: height / 12,
            maxWidth: '95%',
            alignItems: 'center',
            justifyContent: 'center',
        },
    });
    return ( <KeyboardAvoidingView style={styles.container} behavior="padding">{children}</KeyboardAvoidingView>)
}

