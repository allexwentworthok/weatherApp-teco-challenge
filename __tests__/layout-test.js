import React from 'react'
import renderer from  'react-test-renderer'
import Layout from '../src/components/Layout'

test('renders correctly', () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
});
