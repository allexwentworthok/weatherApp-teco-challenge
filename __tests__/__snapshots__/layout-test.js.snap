// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f0f0f0",
          "flex": 2,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  />
</View>
`;
